app/pages/patientdata/patientdata.js

```diff
@@ -504,13 +485,20 @@ export let PatientData = React.createClass({
 
     var refresh = this.props.onRefresh;
     if (refresh) {
-      this.setState({title: this.DEFAULT_TITLE});
-      refresh();
+      this.props.clearPatientData(this.props.routeParams.id);
```

Comment:
Clearing patient data here, allows refresh to result in processing post data being passed in as props as the conditions of componentWillReceiveProps will pass.